services/master/controllers/challengeUpdate.go

```diff
@@ -64,25 +64,34 @@ func ChallengeUpdate(c *fiber.Ctx) error {
 	}
 
 	log.Println("Pull successful for", teamName, ". Building image...")
-
-	// Build the challenge with Dockerfile
+	patchone, err := exec.Command("docker", "inspect", dir).Output()
 	cmd := exec.Command("docker", "build", "-t", dir, "./teams/"+dir)
 	cmd.Run()
-
-	// Create a labelSelector to get the challenge pod
-
-	labelSelector := metav1.LabelSelector{
-		MatchLabels: map[string]string{
-			"app": teamName + "_" + challengeName,
-		},
-	}
-	// Delete the challenge pod
-	err = client.CoreV1().Pods(namespace).DeleteCollection(context.Background(), metav1.DeleteOptions{}, metav1.ListOptions{
-		LabelSelector: metav1.FormatLabelSelector(&labelSelector),
-	})
+	cmd = exec.Command("minikube", "image", "load", dir)
```

Comment:
why is there a minikube command?

Reply:
Ye cheez ashpect ke code mai ho rakhi thi :smiling_face_with_tear: 
Challenge deploy mai  bhi ye ho  rha tha

Reply:
Cause there are two ways to do it, i.e to build an image in minkube registry and use it . First make the registry same, docker and Minikube (eval ... etc.) and then using docker commands builds it inside the minikube registry. OR first build it outside and then push it inside and use it, which is what is done here. Both are essentially the same concept. The issue with eval is that it's short term , we have to run it every time to make the registry same , with each terminal instance. And having a local image in docker helps with accessing the image easily in case u need to, hence went with the second.

Reply:
It was also discussed in the meeting. The third point in this thread provides better explanation in case needed.https://sdslabs.slack.com/archives/CMXQ15U87/p1686080227307769

Reply:
We discuss that we would be mounting the docker registry so I don't think this is needed after that.

Reply:
Opened issue #65 for this to discuss.